geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/repository/IndexRepositoryImpl.java

```diff
@@ -105,9 +107,11 @@ public void update(Object key, Object value) throws IOException {
         stats.incFailedEntries();
         logger.info("Failed to update index for " + value + " due to " + e.getMessage());
       }
-      docs.forEach(doc -> SerializerUtil.addKey(key, doc));
-      Term keyTerm = SerializerUtil.toKeyTerm(key);
-      writer.updateDocuments(keyTerm, docs);
+      if (!docs.isEmpty()) {
```

Comment:
This still seems logically incorrect. The users serializer could return an empty list of documents. In that case we do want call writer.updateDocuments to remove the old docs. But if there is an exception we don't want to do that.